### PR TITLE
Fix expn/exp1/expi returning nan at infinite arguments

### LINKED ISSUE #38519: jax.scipy.special.exp1/expn/expi return NaN at infinite argument (scipy returns the finite limit)
### Description

The exponential-integral family returns `nan` at infinite arguments, where the mathematical limit is finite and SciPy returns it:

```python
import jax, numpy as np
jax.config.update('jax_enable_x64', True)
import jax.scipy.special as jss, scipy.special as ss
print(jss.exp1(np.inf), ss.exp1(np.inf))        # nan  vs 0.0
print(jss.expn(2, np.inf), ss.expn(2, np.inf))  # nan  vs 0.0
print(jss.expi(-np.inf), ss.expi(-np.inf))      # nan  vs -0.0
```

| call | JAX | SciPy |
|---|---|---|
| `exp1(inf)` | nan | 0.0 |
| `expn(1, inf)` / `expn(2, inf)` / `expn(3, inf)` | nan | 0.0 |
| `expi(-inf)` | nan | -0.0 |
| `expi(+inf)` | nan | inf |

Reproduces in float32 and float64. `inf` is a well-defined limit point of these functions (the integrand `e^{-t}/t^n` vanishes), not a singularity — large finite arguments are correct (`exp1(700)=1.4e-307`, matching scipy), only the infinite limit is wrong.

### Root cause

`exp1(x) = expn(1, x)`, so the bug is in `jax/_src/scipy/special.py::expn`. Its branch selection (`jnp.piecewise`) routes `x > 1` to the asymptotic expansion `_expn2`, and **no condition handles `x == inf`**:

```python
conds = [ (n<0)|(x<0), (x==0)&(n<2), (x==0)&(n>=2),
          (n==0)&(x>=0), (n>=5000), (x>1) ]
vals  = [ nan, inf, 1/n1, exp(-x)/x, _expn3, _expn2, _expn1 ]
```

`_expn2(inf)` evaluates the asymptotic series at infinity, where terms like `exp(-x)·poly(x)` become `0·inf` → nan, instead of the limit 0. `expi` has the analogous gap.

### Suggested fix

Add an `x == inf` condition returning `0` (for n ≥ 1) before the `x > 1` branch, and the symmetric handling in `expi` (`+inf → +inf`, `-inf → 0`).

### System info

```
jax: 0.10.1, jaxlib: 0.10.1, numpy 2.4.6, python 3.12.13, CPU (arm64, macOS)
```

### PR DESCRIPTION
**Pull request by Sumu004**
`expn`, `exp1`, and `expi` return `nan` at infinite arguments where the limit is finite and scipy returns it correctly.

`expn`/`exp1` hit an indeterminate `inf * 0` inside the continued-fraction branch even though `lax.select` should mask it — added an explicit `isinf(x) & (x > 0)` case returning `0` (the integrand vanishes as `t -> inf`). `expi(+inf)` hits the same `inf * 0` form in `_eval_expint_k` — wrapped the result in `jnp.where(isposinf(x), inf, result)`. `expi(-inf)` falls out for free once `exp1` is fixed.

Fixes #38519

### CODE REVIEW

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
